suite-native/module-home/src/components/BiometricsBottomSheet.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { Platform } from 'react-native';
+import { useSelector } from 'react-redux';
+import { AuthenticationType } from 'expo-local-authentication';
+
+import { analytics, EventType } from '@suite-native/analytics';
+import { BottomSheet, Box, Button, Text } from '@suite-native/atoms';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import {
+    getSupportedBiometricsTypes,
+    authenticate,
+    useIsBiometricsEnabled,
+    getIsBiometricsFeatureAvailable,
+} from '@suite-native/biometrics';
+import { Translation, useTranslate } from '@suite-native/intl';
+import { Icon } from '@suite-common/icons/src';
+import { useToast } from '@suite-native/toasts';
+import { selectHasDecidedBiometrics, setHasDecidedBiometrics } from '@suite-native/module-settings';
+
+const SHOW_TIMEOUT = 1500;
+
+const cardStyle = prepareNativeStyle(utils => ({
+    alignItems: 'center',
+    textAlign: 'center',
+    borderRadius: utils.borders.radii.medium,
+    backgroundColor: utils.colors.backgroundSurfaceElevation1,
+    paddingVertical: utils.spacings.large,
+    marginBottom: utils.spacings.large,
+    marginTop: utils.spacings.extraLarge,
+    gap: utils.spacings.large,
+}));
+
+const imageWrapperStyle = prepareNativeStyle(utils => ({
+    padding: 12,
+    borderRadius: utils.borders.radii.round,
+    backgroundColor: utils.colors.backgroundSurfaceElevation2,
+    color: utils.colors.iconPrimaryDefault,
+    width: 88,
+    height: 88,
+}));
+
+const textContentStyle = prepareNativeStyle(utils => ({
+    gap: utils.spacings.small,
+}));
+
+const buttonWrapperStyle = prepareNativeStyle(utils => ({
+    gap: utils.spacings.medium,
+    paddingHorizontal: utils.spacings.small,
+}));
+
+type BiometricsBottomSheetProps = {
+    onClose?: (isVisible: boolean) => void;
+};
+
+export const BiometricsBottomSheet = ({
+    onClose = (_: boolean) => {},
+}: BiometricsBottomSheetProps) => {
+    const { applyStyle } = useNativeStyles();
+    const { translate } = useTranslate();
+    const { showToast } = useToast();
+    const hasDecidedBiometrics = useSelector(selectHasDecidedBiometrics);
+    const { isBiometricsOptionEnabled } = useIsBiometricsEnabled();
+
+    const [isVisible, setIsVisible] = useState(false);
+    const [biometricsTypes, setBiometricsTypes] = useState<AuthenticationType[] | undefined>(
+        undefined,
+    );
+
+    const isFace = biometricsTypes?.filter(x =>
+        [AuthenticationType.FACIAL_RECOGNITION, AuthenticationType.IRIS].includes(x),
+    );
+
+    const iconName = useMemo(() => {
```

Comment:
useMemo is probably not neccessary at all because this super simple if will consume same resources as usage of useMemo 

`const iconName = isFace ? "faceId" : "fingerprint"`

Reply:
right, i didnt probably think at that moment.

Reply:
changed  in d385a4d